experimental/addons/redis-operator/README.md

```diff
@@ -0,0 +1,114 @@
+# redis-operator
+
+Redis Operator creates/configures/manages high availability redis with sentinel automatic failover atop Kubernetes.
+
+This addon will give you the ability to use `redis-failover` Components in Applications.
+
+## Quick Start
+
+In this example, we will :
+- start a high-availability Redis cluster with Sentinel
+- scale it up/down
+- use `redis-cli` to connect and read/write some data
+
+Let's get started.
+
+After you enable this addon, apply this yaml:
+
+```yaml
+apiVersion: core.oam.dev/v1beta1
+kind: Application
+metadata:
+  name: redis-operator-sample
+spec:
+  components:
+    # This component is provided by redis-operator addon.
+    # In this example, 2 redis instance and 2 sentinel instance
+    # will be created.
+    - type: redis-failover
+      # Remember this name, we will use it to find the Service to connect.
+      name: ha-redis
+      properties:
+        # You can increase/decrease this later to add/remove instances.
+        replicas: 2
+```
+
+Verify if we have 2 Redis instances available:
+
+```shell
+$ kubectl get pods -l 'app.oam.dev/name=redis-operator-sample'
+NAME                            READY   STATUS    RESTARTS   AGE
+rfr-ha-redis-0                  1/1     Running   0          2m7s
+rfr-ha-redis-1                  1/1     Running   0          2m7s
+rfs-ha-redis-5c49b8947d-lcqhx   1/1     Running   0          2m7s
+rfs-ha-redis-5c49b8947d-vn6gs   1/1     Running   0          2m7s
+```
+
+As we can see, 2 instances are available. `rfr` refers to RedisFailoverRedis. `rfs` refers to RedisFailoverSentinel.
+
+Scale up/down. As an example, we will scale it down to 1:
+
+```yaml
+apiVersion: core.oam.dev/v1beta1
+kind: Application
+metadata:
+  name: redis-operator-sample
+spec:
+  components:
+    - type: redis-failover
+      name: ha-redis
+      properties:
+        # Change this value. For example, we try to scale it down to 1.
+        replicas: 1
+```
+
+After applying this yaml, we should see we only have 1 Redis instance available now:
+
+```shell
+$ kubectl get pods -l 'app.oam.dev/name=redis-operator-sample'
+NAME                            READY   STATUS    RESTARTS   AGE
+rfr-ha-redis-0                  1/1     Running   0          68s
+rfs-ha-redis-7fd59f4758-g6dng   1/1     Running   0          80s
+```
+
+Now, we will try to connect to the Redis cluster (specifically, connect to Sentinel service). To use `redis-cli`, we will attach a terminal to the `redis-cli` Pod so that we hava a `redis-cli` available.
+
+```shell
+# Launch a pod with redis-cli in it.
+kubectl run redis-cli --rm --restart=Never \
+  --image=goodsmileduck/redis-cli -it \
+  --command -- /bin/sh
+```
```

Comment:
check the topology here:

https://github.com/kubevela/catalog/blob/master/experimental/addons/clickhouse/template.cue#L14-L26

Reply:
https://kubevela.io/docs/reference/topology-rule

Reply:
After topology added, the velaux can see the resource topology. What more, you can also use `vela exec/port-forward/logs/ status --endpiont` for the underlying information.

Reply:
Great! It is used now